src/opt.cpp

```diff
@@ -387,9 +387,11 @@ void ispc::Optimize(llvm::Module *module, int optLevel) {
     }
     optPM.addModulePass(llvm::VerifierPass(), 0);
 
+    const bool widenIndVars = !g->target->isXeTarget();
+
```

Comment:
Widening disabled for GPU.

Reply:
Why are you pushing this to this PR? We have enough issues to resolve here, so we cannot converge for three month already. Please take this change separately.

Reply:
Reverting this change will effect the ind-var test for GPU: https://github.com/ispc/ispc/pull/2494#discussion_r1287760094

The previous version checked for the promoted type. Do you prefer to remove the test until the final solution is present?